usr/src/uts/i86pc/os/intr.c

```diff
@@ -471,6 +472,7 @@
 #include <sys/hypervisor.h>
 #endif
 
+boolean_t kernel_preempt_enable = B_FALSE;
```

Comment:
Setting this allows higher-priority kernel threads to preempt lower-priority kernel threads.  For example, it allows mac_soft_ring_worker() to preempt kmem_reap().  This is important because mac_soft_ring_worker() is bound to a CPU, and needs to run to service network traffic, but currently it cannot run until kmem_reap() voluntarily blocks, which can in some cases take dozens of seconds.